csm/common/filter.py

```diff
@@ -0,0 +1,100 @@
+# CORTX-CSM: CORTX Management web and CLI interface.
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+from cortx.utils.data.access.filters import Compare, And, Or
+from schematics.exceptions import ConversionError
+from csm.core.blogic.models import CsmModel
+import urllib.parse
+
+class Filter():
+    """Query to filter_object builder"""
+    @staticmethod
+    def parse_query(fields_of_filter):
+        """
+        Segregate list of key_val pairs and operands.
+        :param fields_of_filter: list of key_val pairs and operands like ['field1=val1', 'operand1', 'field2=val2']
+        :returns: dict(key_val_fields), dictionary of field(key) and value pairs
+        :returns: operations, Ordered list of operations to be performed on give field and value pairs
+        """
+        operations = []
+        key_val_fields = []
+        for key in fields_of_filter:
+            if key:
+                if key == "OR" or key == "AND":
+                    operations.append(key)
+                else:
+                    key_val_fields.append(map(str.strip, key.split('=', 1)))
+        return dict(key_val_fields), operations
+
+    @staticmethod
+    def validate_query_fields(query_fields, model):
+        """
+        Validate and provide default values for those keys whose values are invalid for given field.
+        :param query_fields: collection of key value pairs received as input
+        :param Type[BaseModel] model: model for constructing data mapping for index
+        :returns: Updated key_value pairs
+        """
+        default_val = model()
+        default_val_dict = default_val.to_native()
+        updated_key_val = {}
+        for key, value in query_fields.items():
+            field = eval(f"model.{key}")
+            try:
+                valid_operand = field.to_native(value)
+            except ConversionError as e:
+                updated_key_val[key] = default_val_dict[key]
+        return updated_key_val
+
+    @staticmethod
+    def _prepare_filters(query_rcvd, model):
```

Comment:
_prepare_filters --> prepare_filters .

Remove staticmethod

Reply:
query_rcvd -> query

Reply:
Addressed it by renaming variables.